Add 3d Printing Content

davyhallihan opened the pull request:
Add 3d Printing Content surrounding common printers and filaments, basic skill guides, terminology and knowledge, and troubleshooting. 

Content was originally published on https://www.coreftc.org, and is being provided by the maintainers of that site. (@ineedastim @davyhallihan)

texasdiaz commented:
Wow, this is fantastic content. I'm still wading through it, but would you like to add a "Special Thanks" segment at the bottom of the `/manufacturing/3d_printing/index.html` page similar to the "Special Thanks" at the bottom of this page: https://ftc-docs.firstinspires.org/en/latest/ftc_ml/index.html ?

texasdiaz commented on docs/source/manufacturing/3d_printing/general_knowledge/common_upgrades/images/stealthburner.png:
This image is 707,658 bytes, it needs to be under 500,000 bytes to satisfy our individual-file-size storage requirements.

texasdiaz commented on docs/source/manufacturing/3d_printing/filament_choice/advanced_filaments/images/tpu_intake_rollers.png:
This image is 515,733 bytes, it needs to be under 500,000 bytes to satisfy our individual-file-size storage requirements.

texasdiaz commented on docs/source/manufacturing/3d_printing/3d_printed_parts/images/powerswitchmounting.png:
This image is 506,792 bytes, it needs to be under 500,000 bytes to satisfy our individual-file-size storage requirements.

texasdiaz commented on docs/source/manufacturing/3d_printing/troubleshooting/troubleshooting.rst:
```diff
@@ -0,0 +1,9 @@
+
+Here are some hyperlinks to common issues with printers that are worth looking at during a troubleshooting process.
+
+* `Underextrusion and Overextrusion <../specific_skill_guides/temperature_tuning/temperature_tuning.html>`_
```
This link, and the rest of the links on this page, attempt to directly reference the html directory hierarchy. We try not to do this - direct html linking to internal pages, I mean (this referencing is a form of directly linking, assuming we know where Sphinx will put the files in the final html hierarchy). 

Instead, if you truly want to use direct references, use the :doc: command instead of direct linking (with the underscore) to reference internal 
RST docs using the repo hierarchy. 

As an example, I would rewrite this line as:
```
*  :doc:`Underextrusion and Overextrusion <../specific_skill_guides/temperature_tuning/temperature_tuning>`
```

The remaining links would be very similar.

texasdiaz reviewed (changes requested):
Here are a few quick review notes that jump out at me.

davyhallihan commented:
Few commits after I saw the failed checks and Diaz's comments, here's what got fixed: 

- Added a special thanks section similar to that! 
- The image sizes were fixed, I'm not sure how those 3 slipped through when we were doing mass compression/resizing of the ~60 images we added, but they're good to go now. 
- The spelling checks that were failed were also fixed. 
- Troubleshooting hyperlinks were changed to :doc: references

texasdiaz commented on docs/source/manufacturing/3d_printing/troubleshooting/troubleshooting.rst:
```diff
@@ -3,7 +3,7 @@ Troubleshooting Common Issues
-* `Bed Not Level <../specific_skill_guides/bed_leveling/bed_leveling.html>`_
-* `Parts Not Fitting <../designing_and_slicing/tolerancing/tolerancing.html>`_
\ No newline at end of file
+:doc:`Underextrusion and Overextrusion <../specific_skill_guides/temperature_tuning/temperature_tuning>`
```
You forgot to keep the bullets, and if you no longer want the bullets you need a space between each line to prevent run-on links.

texasdiaz commented on docs/source/conf.py:
```diff
@@ -282,7 +282,7 @@
     print('Building booklets')
     latex_documents = [
-	('ftc_ml/index', "ftc_ml.tex", "FTC Machine Learning", author, "manual"), # FTC ML
+	    ('ftc_ml/index', "ftc_ml.tex", "FTC Machine Learning", author, "manual"), # FTC ML
```
Why did this get indented unnecessarily?

texasdiaz reviewed (changes requested):
I think the outstanding issue is the indentation in the config file.

davyhallihan commented:
> I think the outstanding issue is the indentation in the config file.

I had changed that to standardize the indentation in that array, where we had added the 3d printing booklet at uday's request. It looked to be the correct formatting given all the other arrays in that conf.py. If this broke something or isn't proper let me know, I had just changed it since it looked to be incorrect.

> Are you planning on more changes, or do you want to change this PR from a DRAFT PR to a real PR?

We have no other changes planned for right now, so it can be changed to a real PR if you have nothing else you would like us to revise in the draft stage!

uvidyadharan reviewed (changes requested):
I feel like I just learned a ton of information. I only have two nitpicks I found while I was reading it.
